imports/server/hooks/dingword-hooks.ts

```diff
@@ -0,0 +1,78 @@
+import { Meteor } from 'meteor/meteor';
+import Flags from '../../flags';
+import ChatNotifications from '../../lib/models/chat_notifications';
+import ChatMessages from '../../lib/models/chats';
+import Profiles from '../../lib/models/profiles';
+import Hookset from './hookset';
+
+const DingwordHooks: Hookset = {
+  onChatMessageCreated(chatMessageId: string) {
+    // Respect feature flag.
+    if (Flags.active('disable.dingwords')) {
+      return;
+    }
+
+    // const start = Date.now();
+    const chatMessage = ChatMessages.findOne(chatMessageId)!;
+
+    if (!chatMessage.sender) {
+      // Don't notify for system messages.
+      return;
+    }
+
+    // Find all users who are in this hunt.
+    const huntMembers = Meteor.users.find({
+      hunts: chatMessage.hunt,
+    }, {
+      fields: { _id: 1 },
+    }).fetch().map((u) => {
+      return u._id;
+    });
+
+    // Fetch the profiles of all those users.
+    const memberProfiles = Profiles.find({
+      _id: {
+        $in: huntMembers,
```

Comment:
nit: might be nice to throw `'dingwords.0': {$exists: true}` as a query to capture "has any dingwords" (and you could index on that too)

Reply:
Ahhhhh, nice!  I tried something like that with `$ne: []` but tsc didn't like the types.

I was given to understand that most queries having to do with array fields were not indexable?  but maybe `.0` is special?

Reply:
Arrays are definitely indexable. If you index an array field, then it explodes out the array and creates an index entry for each value (which is a perf nightmare if your arrays get big, but not a risk here). But indexing a single array element shouldn't explode or do anything unusual.

Reply:
Another option would be to recompute a per-hunt dingwords cache each time a user saves their profile, shifting the load from "every chat message" to "every dingword configuration change" (and I do anticipate the latter being much less frequent an operation).  We'd have to do some locking to avoid dropping dingword updates if multiple profile saves are in flight at once though.